dvc/repo/reproduce.py

```diff
@@ -64,7 +64,7 @@ def _get_stage_files(stage: "Stage") -> typing.Iterator[str]:
         if (
             not dep.use_scm_ignore
             and dep.is_in_repo
-            and not stage.repo.dvcfs.isdvc(dep.fs_path)
+            and not stage.repo.dvcfs.isdvc(str(dep))
```

Comment:
Note that we don't use `dep.def_path` since it's relative the stage `wdir`. `dep.__str__` gives us path relative to the DVC repo root.

Reply:
Probably out of the scope of the fix, but feels a little odd to have (and rely on) this logic under a representation method